index.js

```diff
@@ -516,12 +540,39 @@ class KeyringController extends EventEmitter {
    * @param {string} password - The keyring controller password.
    * @returns {Promise<boolean>} Resolves to true once keyrings are persisted.
    */
-  async persistAllKeyrings(password = this.password) {
-    if (typeof password !== 'string') {
+  async persistAllKeyrings(password) {
+    if (password && typeof password !== 'string') {
       throw new Error('KeyringController - password is not a string');
     }
 
-    this.password = password;
+    // MV3: Since we also allow persisting without a password,
+    // we should require this.encryptedKey
+    if (password === undefined && this.encryptedKey === undefined) {
+      return Promise.reject(
+        new Error(
+          'KeyringController - a password or encryptedKey must exist to persist keyrings',
+        ),
+      );
+    }
```

Comment:
Throwing inside an async function has the same result of rejecting the promise returned by it. 
Is this an optimization? Was throwing noticably slower?